openmls/src/ciphersuite/mod.rs

```diff
@@ -569,9 +636,10 @@ pub struct Ciphersuite {
     name: CiphersuiteName,
     signature_scheme: SignatureScheme,
     hpke: Hpke,
-    aead: AeadMode,
-    hash: DigestMode,
-    hmac: HmacMode,
+    // The rng really must not be accessed concurrently
+    rng: Mutex<ChaCha20Rng>,
```

Comment:
Having the state here is probably the right place. I'm however wondering about the following: in the past we used to instantiate new `Ciphersuite` objects from a `CiphersuiteName`. This was not an issue, because there was no state but this changes now. Did you already check if we run into trouble because of that?

Reply:
This will just take the same `Ciphersuite` object and hence the same randomness pool. Which is fine from a security perspective but of course adds a single mutex for everything using this cipher suite.
This is something to consider in #467 and #468